docs/products/eigenda/core-concepts/payments.md

```diff
@@ -19,53 +23,81 @@ The overall goal of the payments upgrade is to introduce flexible payment modali
+To initiate a dispersal, the EigenDA client sends a dispersal request containing a payment header to the disperser,
+which validates the payment information. For on-demand payments, the disperser tracks usage and validates against
+deposits in the PaymentVault contract. For reservation payments, validator nodes serve as the source of truth, tracking
+each account's reservation usage using leaky bucket rate limiting. Clients can query the disperser to retrieve their own
+offchain state for on-demand usage information.
 
 ## Low-level Specification
 
 ### On-Demand Bandwidth (On-Demand Payments)
 
-Requests created by the disperser client contain a `BlobHeader`, which contains a `PaymentMetadata`  struct as specified below. 
+On-demand payments are supported only through the EigenDA Disperser, which tracks usage and validates payments.
+
+Requests created by the disperser client contain a `BlobHeader`, which contains a `PaymentMetadata` struct as specified
+below. 
 
 ```go
 // PaymentMetadata represents the payment information for a blob
 type PaymentMetadata struct {
   // AccountID is the ETH account address for the payer
   AccountID string
-  // Timestamp represents the nanosecond of the dispersal request creation
-  // ! this field is not used for on-demand payments
+  // Timestamp represents the nanosecond of the dispersal request creation (serves as nonce)
   Timestamp int64
-  // CumulativePayment is the amount of payment cumulated for all previous and current dispersal
+  // CumulativePayment represents the total amount of payment (in wei) made by the user up to this point.
+  // If empty/zero → reservation payment
+  // If non-zero → on-demand payment
   CumulativePayment *big.Int
 }
 ```
 
-On-demand bandwidth users must first deposit tokens into the payment vault contract for a particular account, in which the contract stores the total payment deposited to that account (`totalDeposit`). Users should be mindful in depositing as they cannot withdrawal or request for refunds from the current Payment Vault contract. This amount will be validated against the payment metadata's `CumulativePayment` field authroized by the user for each blob dispersal request. The network will ensure that the `CumulativePayment` is great enough to cover the bandwidth usage by the user but still remains with the total amount deposited on chain. 
+On-demand bandwidth users must first deposit tokens into the payment vault contract for a particular account, in which
+the contract stores the total payment deposited to that account (`totalDeposit`). Users should be mindful in depositing
+as they cannot withdrawal or request for refunds from the current Payment Vault contract.
```

Comment:
not sure if you mention this elsewhere but might be good to mention that a user can query the disperser to determine its current remaining balance